### PR TITLE
Background writeup for trees and graphs

### PR DESCRIPTION
**Pull request by etcadinfinitum**
Because some attendees are not at the 300-level courses yet (specifically 343) or might benefit from a refresher on these data structures, I am creating a background section for these two concepts. When I have added the remaining segments, I'll mark this as ready for review.

### CODE REVIEW

File: Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md [hunk 1]
```diff
@@ -76,6 +76,40 @@ An example of a binary search tree that contains characters is:
 
 ### What is a graph?
 
+A graph is also a data structure that organizes and shows relationships 
+between individual data nodes. However, a graph is not organized in a 
+heirarchal way the way a tree is. Nodes in a graph are not required
```
Comment (bpas247) [inline Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md]: Might be worth noting that a Tree is generally just an undirected graph.

File: Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md [hunk 2]
```diff
@@ -76,6 +76,40 @@ An example of a binary search tree that contains characters is:
...
+
+#### Undirected Graphs
+
+![facebook as an undirected graph](./background/facebook-undirected.png)
+
+#### Directed Graphs
+
+![twitter as a directed graph](./background/twitter-directed.png)
+
+#### Weighted Graphs
```
Comment (bpas247) [inline Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md]: Are there problems that will be presenting on weighted graphs? This section might not be needed if we aren't.

Comment (etcadinfinitum) [reply Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md]: I will put some more thought into it. I have a (perhaps overly ambitious) plan to do Kevin Bacon using a weighted graph. I'll get back to you on this one

File: Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md (hunk 1 above)
Comment (etcadinfinitum) [reply Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md]: Addressed in 704a50576a2bded0c083637c42955e7fa328bd69

File: Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md (hunk 2 above)
Comment (etcadinfinitum) [reply Spring-2019/trees_graphs/SOLUTIONS_2019Spring.md]: So, I'm giving the problem as a weighted graph, so I'm inclined to stick with this section in the background section :grin:

**Pull request merged**